third_party/terraform/resources/resource_cloudfunctions_function.go

```diff
@@ -166,6 +182,22 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"ingress_settings": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  functionDefaultIngressSettings,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
Thanks for adding the validation for the field! We actually have a commonly used one at `validation.StringInSlice` that should have the same effect.

Reply:
ok i will look into it

Reply:
Thanks for the advice, the function you mentioned is a lot better. I will update the pr